### PR TITLE
Adds option that allows passing component tags as reference

### PR DESCRIPTION
**Pull request by mairatma**
I'm not sure if this is something that you'd want to merge, but I added an option on my fork of babel-incremental-dom to enable passing upper case tags by reference to incremental dom calls, even though that doesn't work without extra code. I need this for a project I'm working on, where we already handle this case internally, but needed the compiler to work as expected.

I saw that you already have an issue (https://github.com/jridgewell/babel-plugin-incremental-dom/issues/9) for working on this feature, but since it seems to waiting for some changes on incremental dom's side I think this option I've added could be useful for some people, so sending it in case you agree as well :)

### CODE REVIEW

File: src/helpers/element-close-call.js [hunk 1]
```diff
@@ -18,5 +18,6 @@ export default function elementCloseCall(t, path, plugin) {
     }
   }
 
-  return toFunctionCall(t, iDOMMethod("elementClose", plugin), [toReference(t, node.name)]);
+  const isComponent = /^[A-Z]/.test(node.name.name);
```
Comment (jridgewell) [inline src/helpers/element-close-call.js]: Can you create a helper specifically for this? `isComponent` sounds perfect.

File: README.md [hunk 2]
```diff
@@ -192,6 +192,38 @@ plugin:
...
+```
+
+To do this, simply add the `componentAsReference` option to the Incremental DOM
+plugin:
+
+```json
+{
+  "plugins": [[
+    "incremental-dom", {
+      "componentAsReference": true
```
Comment (jridgewell) [inline README.md]: Let's just call this `components`.

**Comment by jridgewell** [comment]
Yay contributions! I think the only thing missing is a test for `<namespace.Component />`.

**Comment by mairatma** [comment]
Thanks for considering and reviewing this! I've added the changes you've requested, so let me know if they're what you meant :)

File: src/helpers/element-close-call.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ import toReference from "./ast/to-reference";
 import iDOMMethod from "./idom-method";
 
 // Returns the closing element's function call.
-export default function elementCloseCall(t, path, plugin) {
+export default function elementCloseCall(t, path, plugin, { components, isComponent }) {
```
Comment (jridgewell) [inline src/helpers/element-close-call.js]: Almost, I was aiming for a `isComponent` helper function like `iDOMMethod` and similar:

```js
# helpers/is-component.js
export default function isComponent(tag) {
  return /^[A-Z]/.test(tag);
}

# helpers/element-close-call.js
import isComponent from './is-component';

//...
```

**Comment by mairatma** [comment]
Makes sense, done :)

**Comment by jridgewell** [comment]
Thanks!

**Comment by mairatma** [comment]
Thank you! Could you also release an npm version with these changes so I can use them instead of my fork? :)

**Comment by jridgewell** [comment]
I'll try to get to that tomorrow.

**Comment by jridgewell** [comment]
Published v3.1.0

**Comment by anilanar** [comment]
@mairatma What kind of extra code is needed to be able to use this?

**Comment by mairatma** [comment]
@anilanar you just need to make sure that you're using `babel-plugin-incremental-dom` version >= 3.1.0 when building your code.

**Pull request merged**